xfsprogs.yaml

```diff
@@ -217,6 +224,52 @@ subpackages:
             mkfs.xfs xfs.img
             xfs_admin -L "TestXFS" xfs.img
             xfs_admin -l xfs.img | grep 'TestXFS'
+        - name: "Test xfs reproducibility"
+          runs: |
+            dd if=/dev/zero of=xfs1.img bs=1M count=300
+            dd if=/dev/zero of=xfs2.img bs=1M count=300
+            SOURCE_DATE_EPOCH=1744654025
+            export LD_PRELOAD="/usr/local/lib/faketime/libfaketime.so.1"
+            export FAKERANDOM_SEED="0x12345678DEADBEEF"
+            export FAKETIME="$(date -d @"$SOURCE_DATE_EPOCH" +'%Y-%m-%d %H:%M:%S')"
+            export NO_FAKE_STAT=1
+            mkfs.xfs \
+              -b size=4096 \
+              -d agcount=4 \
+              -d noalign \
+              -i attr=2 \
+              -i projid32bit=1 \
+              -i size=512 \
+              -l size=67108864 \
+              -l su=4096 \
+              -l version=2 \
+              -m crc=1 \
+              -m finobt=1 \
+              -m uuid=44444444-4444-4444-4444-444444444444 \
+              -n size=16384 \
+              -n version=2 \
+              -p file=/usr \
+              xfs1.img
+            mkfs.xfs \
+              -b size=4096 \
+              -d agcount=4 \
+              -d noalign \
+              -i attr=2 \
+              -i projid32bit=1 \
+              -i size=512 \
+              -l size=67108864 \
+              -l su=4096 \
+              -l version=2 \
+              -m crc=1 \
+              -m finobt=1 \
+              -m uuid=44444444-4444-4444-4444-444444444444 \
+              -n size=16384 \
+              -n version=2 \
+              -p file=/usr \
+              xfs2.img
+              hash1="$(md5sum xfs1.img | cut -d' ' -f1)"
+              hash2="$(md5sum xfs2.img | cut -d' ' -f1)"
```

Comment:
if 'md5sum' isnt present, or fails, this silently assigns empty string to 'hash1' and 'hash2', which then will pass the test below.

you need `set -o pipefail` or just separate lines.

or easier, just use `cmp xfs1.img xfs2.img`